appcues/src/main/java/com/appcues/analytics/ExperienceLifecycleTracker.kt

```diff
@@ -39,18 +37,18 @@ internal class ExperienceLifecycleTracker(
     // AnalyticsTracker -> this <- Analytics Tracker
     private val analyticsTracker: AnalyticsTracker by inject()
     private val storage: Storage by inject()
-    private val appcuesCoroutineScope: AppcuesCoroutineScope by inject()
+    private val appcuesCoroutineScope: CoroutineScope by inject()
     private val config: AppcuesConfig by inject()
 
     private var stateJob: Job? = null
     private var errorJob: Job? = null
 
-    fun start(stateMachine: StateMachine, onEndedExperience: (Experience) -> Unit, dispatcher: CoroutineDispatcher = Dispatchers.IO) {
+    fun start(stateMachine: StateMachine, onEndedExperience: (Experience) -> Unit) {
         // ensure any existing observers are stopped before starting new ones
         stop()
 
-        stateJob = appcuesCoroutineScope.launch(dispatcher) { stateMachine.observeState(onEndedExperience) }
-        errorJob = appcuesCoroutineScope.launch(dispatcher) { stateMachine.observeErrors() }
+        stateJob = appcuesCoroutineScope.launch { stateMachine.observeState(onEndedExperience) }
+        errorJob = appcuesCoroutineScope.launch { stateMachine.observeErrors() }
```

Comment:
removed the `launch(Dispatchers.IO)` here as it causes the tests to hang and not complete analytics handling as expected when running in Robolectric. It seems like it is not really needed? I'm not sure, but the actual network calls that happen later are already using `Dispatchers.IO`.